models/demos/falcon40b/tt/falcon_model.py

```diff
@@ -165,14 +165,8 @@ def model_preprocessing(self, llm_mode, input_ids, kv_cache_len, num_input_token
             attention_mask_bool = torch.ones(batch_size, 1, sequence_size, sequence_size, dtype=bool)
             attention_mask_bool = attention_mask_bool.triu(diagonal=1)
 
-            attention_mask_heads_dim = (
-                self.config.num_attention_heads
-                if self.model_config["ATTN_MASK_MEMCFG"].is_sharded()
-                else len(self.devices)
-            )
-
             attention_mask_bool_chunks = torch.chunk(
-                (attention_mask_bool * -100000).expand(-1, attention_mask_heads_dim, -1, -1),
+                (attention_mask_bool * -100000).expand(-1, len(self.devices), -1, -1),
```

Comment:
Did you test this? Currently it's bfp8. The values are 0 or -100000 in that case. Would any type of mask containing 0s and -100000s result in the same bfp4_b and bfp8_b values? Like: a block of 16x16 0s / -100000s / and a mixture of both?

Reply:
Would be worth trying trying out. I guess if it's working on BFP8 it should also work on BFP4

Reply:
With bfp4 I get good pcc and no hang for S=128, but a hang (1st attempt) and an ARC scratch error (2nd attempt) for S=2048. I'll use bfp8 for now but create an issue for this since it seems to be functionally correct in the S=128 case.